### ISSUE TITLE
svelte-add postcss overwrites previous sveltePreprocess() options

### ISSUE THREAD

**Issue by Leftium** [svelte-add/postcss#17]
**Repro steps:**

1. `npm init svelte@next`
2. Edit svelte.config.cjs so sveltePreprocess looks like this:
    `preprocess: sveltePreprocess({ coffeescript: { bare: true } }),`
3. `npx svelte-add postcss`
4. Inspect svelte.config.cjs:

![image](https://user-images.githubusercontent.com/381217/113604256-9e9f2000-960a-11eb-98d5-826d1b060c20.png)

Modifying sveltePreprocess() is a common, yet error-prone process. I think a robust method of adding/merging sveltePreprocess  into svelte.config.cjs is called for. Perhaps as a small NPM package?

Or perhaps instead of using sveltePreprocess() auto mode, the [stand-alone mode](https://github.com/sveltejs/svelte-preprocess/blob/main/docs/preprocessing.md#difference-between-the-auto-and-stand-alone-modes) should be used?

**Comment by babichjacob** [comment]
Step 2. is not allowed. You may not modify the default template before running any Svelte adders.

**Comment by babichjacob** [comment]
#16 was closed in favor of this. 

I do not currently expect this to work yet, though. I think we need a more advanced command composition system which I won't have enough free time to work on until school is over at the end of the month.

**Comment by babichjacob** [comment]
Would it be possible to share code by exporting common methods from the svelte-adders package? Like this:

    // preset.ts

    import { adder } from "svelte-adders"

    adder.editSvelteConfig()
        .addPreprocessor('sveltePreprocess', {
            defaults: {
                style: "postcss",
            },
            postcss: true
        })

This example mirrors the [API for Preset's `editnodepackages()`](https://usepreset.dev/docs/actions/edit-json/#api-of-editnodepackages). One nice advantage: options can be passed as a true JS object instead of as a string.

Package `svelte-adders` could either wrap/extend package `apply` or work side-by-side. In the side-by-side case, editSvelteConfig() would be called inside `Preset.edit(["svelte.config.cjs"]).update((content)...`, inputting `content` and returning the new string.

I think it is possible to detect the context (imported as a module vs running as node script) so the common functionality could just be added to the existing `svelte-adders` package. If not, a new package could be created.

_Originally posted by @Leftium in https://github.com/svelte-add/svelte-adders/issues/13#issuecomment-813550485_

**Comment by babichjacob** [comment]
> Package `svelte-adders` could either wrap/extend package `apply` or work side-by-side. In the side-by-side case, editSvelteConfig() would be called inside `Preset.edit(["svelte.config.cjs"]).update((content)...`, inputting `content` and returning the new string.


Yes, forking `use-preset` so we can have extended functionality that may or may not be submitted back upstream is the current path of least friction.

But I think a lot of technical debt has already accrued so ditching it (unfortunately) in favor of built-in Node libraries (plus something like `recast` or one of the many JavaScript parsing + stringifying libraries pngwn thankfully informed me of) might be a better idea than that, especially since that will allow sharing code between adders.

**Comment by babichjacob** [comment]
This will be solved in a future rewrite of `svelte-add`: https://github.com/svelte-add/svelte-adders/issues/12

**Comment by babichjacob** [comment]
Don't be confused by the `wontfix` label. It just signals that I can't fix this in the current state of things. Wait like 2-3 weeks for the rewrite (I'm really busy for the next week).

**Comment by Leftium** [comment]
Please support the entire svelte.config.js with the rewrite. The coffeescript adder needs to add a vite plugin, as well:

```

const coffeescript = require('rollup-plugin-coffee-script');

/** @type {import('@sveltejs/kit').Config} */
module.exports = {
    preprocess: [
        sveltePreprocess({
                "coffeescript": {
                    "bare": true
                }
            }),
    ],
    kit: {
        // By default, `npm run build` will create a standard Node app.
        // You can create optimized builds for different platforms by
        // specifying a different adapter
        adapter: netlify(),

        // hydrate the <div id="svelte"> element in src/app.html
        target: '#svelte',

        vite: {
            ssr: {
                noExternal: Object.keys(pkg.dependencies || {})
            },
            plugins: [
                coffeescript()
            ]
        }
    }
};

```

**Comment by babichjacob** [comment]
Thanks! We can definitely do that. I will transfer this issue to the main `svelte-add` repo.
